src/main/java/com/foodybuddy/service/impl/OrderDishServiceImpl.java

```diff
@@ -0,0 +1,155 @@
+		Session session = null;
+		try {
+			if (orderId < 0) {
+				throw new Exception("invalid orderId ");
+			}
+			session = this.sessionFactory.openSession();
+			OrderDishDAO orderDishDAO = new OrderDishDAOImpl(session);
+			List<OrderDish> orderDishList = orderDishDAO.getListByOrderId(orderId);
+			return orderDishList;
+		} catch (Exception ex) {
+			throw new Exception("Listingby Orderid failed: " + ex.getMessage(), ex);
+		} finally {
+			if (session != null) {
+				session.close();
+			}
+		}
+	}
+	public OrderDish insertOrderDish(Order order, Dish dish) throws Exception{
+		sessionFactory = SessionFactoryUtils.getSessionFactory();
+		Session session = null;
+		Transaction transaction = null;
+		try{
+			if(order == null)
+				throw new Exception("invalid order");
+			if(dish == null)
+				throw new Exception("invalid Dish");
+			session = this.sessionFactory.openSession();
+			transaction = session.beginTransaction();
+			transaction.setTimeout(10);
+
+			OrderDishDAO orderDishDAO = new OrderDishDAOImpl(session);
+			OrderDish orderDish = new OrderDish();
+			orderDish.setDish(dish);
+			orderDish.setOrder(order);
+			orderDishDAO.insert(orderDish);
+			session.getTransaction().commit();
+			return orderDish;
+		} catch (Exception ex) {
+			try {
+				if (transaction == null) {
+					throw new Exception("Transaction is null " + ex.getMessage(), ex);
+				}
+				transaction.rollback();
+				throw new Exception("Transaction could not be completed will be rollbacked: " + ex.getMessage(), ex);
+			} catch (Exception rbe) {
+				throw new Exception("Transaction could not be completed and rollback failed: " + ex.getMessage(), ex);
+			}
+		} finally {
+			if (session != null) {
+				session.close();
+			}
+		}
+	}
+
+	public OrderDish updateOrderDishQuantity(int orderDishId, int updatedQuantity) throws Exception {
+		sessionFactory = SessionFactoryUtils.getSessionFactory();
+		Session session = null;
+		Transaction transaction = null;
+		try {
+			if (orderDishId <= 0) {
+				throw new Exception("invalid orderDishId");
+			}
+			if (updatedQuantity < 0) {
+				throw new Exception("invalid updated quantity");
+			}
+
+			session = sessionFactory.openSession();
+			transaction = session.beginTransaction();
+			transaction.setTimeout(10);
+			OrderDish orderDish = new OrderDish();
+			OrderDishDAO orderDishDAO = new OrderDishDAOImpl(session);
+			try{
+				orderDish = orderDishDAO.getById(orderDishId);
+			} catch (Exception ex) {
+				throw new Exception("OrderDish id is invalid" + ex.getMessage(), ex);
+			}
+			
+			Dish dish = orderDish.getDish();
+
+			// Check if available quantity is greater than updated quantity
+			if (dish.getQuantityAvailable() > updatedQuantity) {
+				orderDish.setQuantity(updatedQuantity);
+				dish.setQuantityAvailable(dish.getQuantityAvailable() - updatedQuantity);
+				DishDAO dishDAO = new DishDAOImpl(session);
+				dishDAO.update(dish);
+				orderDishDAO.update(orderDish);
```

Comment:
Test this function. Is it doing cascading ? I have tested that session.persist() is doing cascading .